sklearn/datasets/_arff_parser.py

```diff
@@ -376,6 +377,36 @@ def _pandas_arff_parser(
     columns_to_keep = [col for col in frame.columns if col in columns_to_select]
     frame = frame[columns_to_keep]
 
+    # `pd.read_csv` only consider double quotes as delimiters for strings.
+    # In case that a single quote is used as a delimiter, we need to strip it.
+    pattern = r"^'(?P<contents>.*)'$"
```

Comment:
We might want to precompile this regexp instead of doing it for each categorical column.

Also let's use a more descriptive variable name:

```suggestion
    single_quote_pattern = re.compile(r"^'(?P<contents>.*)'$")
```

Reply:
You might need to add the `import re` statement at the top of the file to accept this suggestion (untested).